Documentation/Contributors/DocumentationGuide/README.md

```diff
@@ -0,0 +1,491 @@
+ * @param {Matrix4} [options.modelMatrix=Matrix4.IDENTITY] The 4x4 matrix that defines the reference frame, i.e., origin plus axes, to visualize.
+ * @param {Boolean} [options.show=true] Determines if this primitive will be shown.
+ * @param {Object} [options.id] A user-defined object to return when the instance is picked with {@link Scene#pick}
+```
+generates
+
+![](DebugModelMatrixPrimitive.png)
+
+If all `options` properties are optional, also mark the `options` object optional.
+
+## Exceptions
+
+* Document exceptions after the `@param`/`@returns` section using `@exception`, e.g.,
+```javascript
+/**
+ * ...
+ *
+ * @exception {DeveloperError} aspectRatio must be greater than zero.
+ */
+Matrix4.computePerspectiveFieldOfView = function(fovY, aspectRatio, near, far, result) {
+    //>>includeStart('debug', pragmas.debug);
+    if (aspectRatio <= 0.0) {
+        throw new DeveloperError('aspectRatio must be greater than zero.');
+    }
+    // ...
+```
+* Do not document exceptions for missing parameters; it is implicit in the `@param` tag because the parameter is not optional, e.g.,
+```javascript
+/**
+ * Computes a Matrix4 instance from a column-major order array.
+ *
+ * @param {Number[]} values The column-major order array.
+ * @param {Matrix4} [result] The object in which the result will be stored, if undefined a new instance will be created.
+ * @returns {Matrix4} The modified result parameter, or a new Matrix4 instance if one was not provided.
+ */
+Matrix4.fromColumnMajorArray = function(values, result) {
+    //>>includeStart('debug', pragmas.debug);
+    if (!defined(values)) {
+        throw new DeveloperError('values is required.');
+    }
+    //>>includeEnd('debug');
+
+    return Matrix4.clone(values, result);
+};
+```
+
+## Examples
+
+Developers almost always jump to an example before reading the doc.  Provide concise but instructive code examples with enough context whenever possible.
+
+Useful examples:
+```javascript
+/**
+ * ...
+ *
+ * @example
+ * var n = Cesium.Math.lerp(0.0, 2.0, 0.5); // returns 1.0
+ */
+CesiumMath.lerp = function(p, q, time) {
+    // ...
+```
+
+```javascript
+/**
+ * ...
+ *
+ * @example
+ * // Apply non-uniform scale to node LOD3sp
+ * var node = model.getNode('LOD3sp');
+ * node.matrix = Cesium.Matrix4.fromScale(new Cesium.Cartesian3(5.0, 1.0, 1.0), node.matrix);
+ */
+Model.prototype.getNode = function(name) {
+    // ...
+```
+
+Unnecessary example:
+```javascript
+/**
+ * ..
+ *
+ * @example
+ * var f = Cesium.Math.EPSILON1;
+ */
+CesiumMath.EPSILON1 = 0.1;
+```
+
+* Use the Cesium namespace (`Cesium.`) in examples.
+* Limit code in `@example` tags to 80 lines so it does not overflow.
+
+## References
+
+* Use `@see` sparingly to link to related classes, functions, and online resources., e.g.,
+```javascript
+/**
+ * A {@link TerrainProvider} that produces terrain geometry by tessellating height maps
+ * retrieved from an ArcGIS ImageServer.
+ *
+ * ...
+ *
+ * @see TerrainProvider
```

Comment:
I don't think this is a good example of `@see` usage because the type is already linked to in the description (so the see is redundant).  Maybe using the TerrainProvider doc itself would be better, since it links to the implementations?

```
    /**
     * Provides terrain or other geometry for the surface of an ellipsoid.  The surface geometry is
     * organized into a pyramid of tiles according to a {@link TilingScheme}.  This type describes an
     * interface and is not intended to be instantiated directly.
     *
     * @alias TerrainProvider
     * @constructor
     *
     * @see EllipsoidTerrainProvider
     * @see CesiumTerrainProvider
     * @see ArcGisImageServerTerrainProvider
     */
    var TerrainProvider = function() {
```